platform/linux-generic/arch/x86/odp_time_cpu.c

```diff
@@ -15,24 +14,67 @@
 
 #include <time.h>
 
-/* Measure TSC frequency. Frequency information registers are defined for x86,
- * but those are often not enumerated. */
+static int nsleep(uint64_t nsec)
+{
+	struct timespec ts1, ts2;
+	uint64_t diff;
+
+	if (clock_gettime(CLOCK_MONOTONIC_RAW, &ts1)) {
+		_ODP_ERR("clock_gettime() failed\n");
```

Comment:
Could be useful to have the actual error printed as well, e.g. with `strerror()`.

Reply:
Added in v2.